.ci/packer_cache.sh

```diff
@@ -15,5 +15,30 @@ while [ -h "$SCRIPT" ] ; do
   fi
 done
 
-source $(dirname "${SCRIPT}")/java-versions.properties
-JAVA_HOME="${HOME}"/.java/${ES_BUILD_JAVA} ./gradlew resolveAllDependencies --parallel
+cd $(dirname "${SCRIPT}")/..
+
+resolveAllDependencies() {
+    local REFRESH_FAILED=no
+    local MAX_TRIES=3
+    for i in $(seq 1 $MAX_TRIES) ; do 
+        echo "Resolving dependencies try $i/$MAX_TRIES"
+        JAVA_HOME="${HOME}"/.java/${ES_BUILD_JAVA} ./gradlew resolveAllDependencies --parallel && break
+        REFRESH_FAILED=yes
+    done
+    if [ $REFRESH_FAILED != "no" ] ; then
+        echo "Resolving dependencies failed after 3 retries ..."
+        exit 1
+    fi   
+}
+
+for branch in "master" "6.x" "5.6" ; do 
```

Comment:
What about 6.4? Are we going to have to maintain this as we bump versions. We should find a better approach. Can this be handled externally, in the packer scripts which have access to the current release versions?

Reply:
6.4 will probably be close to 6.x, but doesn't hurt adding it, it there aren't many differences it will finish quickly and if they are then it was a good idea to add it. 

This does require maintenance on version bumps and I agree 100% it's note ideal. 
I see this as a stop-gap solution so we don't have to time right now to deal with some of these network related build failures. I see this as muting a test. 
The only real solution is the one outlined in the linked ticket: a caching proxy. 
No matter how clever we get with this, we can only build the images so often, we can always have something that won't be on the image when we bump a version of some dependency and run into this issue when it builds on CI.

I don't think we have version specific CI images, or if we need to, so not sure packer should ever be aware of  our versions.  I see this as a simpler approach for a stop-gap.

Reply:
I think it is sufficient to not include 6.4, if we make `resolveAllDependencies` tasks inside the bwc projects under distribution. This way, whatever branches are relevant for bwc will be checked out and the recursive dep resolve will happen.

Overall, though, I'm very apprehensive about this approach...

Reply:
@rjernst I share your sentiment. 
If we'd rather deal with the network failures than do this, I'm fine with that, my intention was to do the minimum to ease the pain until we'll have the cache. 
I also taught about the approach you suggest, but we'll have to do all the checkouts to be able to do so further increasing the image build time ( I know form infra they are concerned about that ). It would make more sense if we were able to keep the checkouts in the image so we only have to update when the build runs.